src/wallet/wallet.cpp

```diff
@@ -4124,7 +4237,8 @@ bool CWallet::NewKeyPool()
         for (int i = 0; i < nKeys; i++)
         {
             int64_t nIndex = i+1;
-            walletdb.WritePool(nIndex, CKeyPool(GenerateNewKey()));
+            auto key = GenerateNewKey();
+            walletdb.WritePool(nIndex, CKeyPool(key));
```

Comment:
I'm confused why this change is being made; it seems redundant.

Separately, we need to think through how the "keypool" should behave now that we have deterministic key generation. Please open an issue for looking into this.

Reply:
I had initially gone through the process of trying to remove the key pool, but it turns out that it's relatively major surgery and I didn't want to make this PR any bigger than it already is. Since these are only accessed by APIs that will be deprecated after the advent of UAs anyway, I figured that they could remain stable so long as GenerateNewKey is doing the right thing. The delta here is just a vestigial bit of that change that I didn't realize I needed to revert.